client/src/components/Meetup/Meetup.js

```diff
@@ -1,8 +1,689 @@
+    } else if (
+      meetupData.pendingAttendees &&
+      meetupData.pendingAttendees.includes(userData.id)
+    ) {
+      setUserMeetupStatus("pending");
+    } else if (
+      meetupData.attendees &&
+      meetupData.attendees.includes(userData.id)
+    ) {
+      setUserMeetupStatus("attending");
+    } else if (meetupData.rejected) {
+      setUserMeetupStatus("rejected");
+    }
+  };
+
+  useEffect(async () => {
+    const { res: userRes, resJSON: userJSON } = await fetcher.getUserData();
+    if (!userRes.ok) {
+      setIsLoading(false);
+      return;
+    }
+    const { res: meetupRes, resJSON: meetupJSON } = await fetcher.getMeetup(id);
+    if (!meetupRes.ok) {
+      setFetchMeetupError(meetupRes.status);
+      setIsLoading(false);
+      return;
+    }
+    const { res: ownerRes, resJSON: ownerJSON } = await fetcher.getUserData(
+      meetupJSON.owner
+    );
+    if (!ownerRes.ok) {
+      setIsLoading(false);
+      return;
+    }
+
+    setUser(userJSON);
+    setEventDetails({
+      description: meetupJSON.description,
+      location: meetupJSON.location,
+      maxCapacity: meetupJSON.maxCapacity,
+      minCapacity: meetupJSON.minCapacity,
+      owner: ownerJSON,
+      tags: meetupJSON.tags,
+      time: new Date(meetupJSON.time),
+      title: meetupJSON.title,
+    });
+    setAttendeeLists(meetupJSON.attendees, meetupJSON.pendingAttendees);
+    setUserStatus(userJSON, meetupJSON);
+    setIsLoading(false);
+  }, []);
+
+  const Spinner = (
+    <Grid
+      container
+      spacing={3}
+      justify="center"
+      alignItems="center"
+      className={classes.spinner}
+    >
+      <CircularProgress />
+    </Grid>
+  );
+
+  const renderError = (status) => {
+    let errorMessage = "";
+
+    switch (status) {
+      case 404:
+        errorMessage = "Error: Meetup not found.";
+        break;
+      case 400:
+        errorMessage = "Error: Bad request.";
+        break;
+      default:
+        errorMessage = "Error: Unspecified error occured.";
+    }
+
+    return (
+      <Grid container spacing={1}>
+        <Typography className={classes.error}>{errorMessage}</Typography>
+      </Grid>
+    );
+  };
+
+  const renderTags = (tags) => {
+    // TODO: convert to search link w/ tags
+    return (
+      <>
+        {tags.map((tagText) => (
+          <li key={tagText} className={classes.tag}>
+            <Chip
+              clickable
+              size="small"
+              label={tagText}
+              // component={RouterLink}
+              // to=""
+            />
+          </li>
+        ))}
+      </>
+    );
+  };
+
+  const renderLocation = () => {
+    let locationLink;
+    if (eventDetails.location.url) {
+      const link = (
+        <>
+          (
+          <Link
+            href={eventDetails.location.url}
+            rel="noreferrer"
+            target="_blank"
+          >
+            link
+          </Link>
+          )
+        </>
+      );
```

Comment:
```suggestion
      const link = (
          <Link
            href={eventDetails.location.url}
            rel="noreferrer"
            target="_blank"
          >
            link
          </Link>
      );
```

Reply:
this was to get parentheses around the link that aren't also part of the link.. not sure if there's any better way to do it